lib/smb2/tree.rb

```diff
@@ -28,23 +34,68 @@ def initialize(client:, tree_connect_response:)
   # The protocol is persnickety about format. The `filename` must not begin
   # with a backslash or it will return a STATUS_INVALID_PARAMETER.
   #
+  # If the optional code block is given, it will be passed the opened file as
+  # an argument and the {File} object will automatically be {File#close
+  # closed} when the block terminates. The value of the block will be
+  # returned.
+  #
   # @param filename [String,#encode] this will be encoded in utf-16le
+  # @param mode [String] See stdlib ::File#mode
+  # @yield [Smb2::File]
   # @return [Smb2::File]
-  def create(filename)
-    packet = Smb2::Packet::CreateRequest.new do |request|
-      request.filename = filename.encode("utf-16le")
-      # @todo document all these flags. value copied from smbclient traffic
-      request.desired_access = 0x0012_0089
-      request.impersonation = 2
-      request.share_access = 3  # SHARE_WRITE | SHARE_READ
-      request.disposition = 1   # Open (if file exists open it, else fail)
-      request.create_options = 0x40 # non-directory
-    end
+  def create(filename, mode = "r+")
```

Comment:
Because `create` has different `return` based on whether there is a block given or not you should use `@overload` to document the behavior with and without a block separately.  In addition to `@yield`, you should use `@yieldparam` and `@yieldreturn` since the value returned from the block is captured.  The format for `@yield` in relation to `@yieldparam` is `@yield [param1, param1]` and then those param names are used in `@yieldparam param1 [Type] description`.